// op-viem/dist/cjs/decorators/walletL1OpStackActions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.walletL1OpStackActions = void 0;
const writeDepositERC20_js_1 = require("../actions/wallet/L1/writeDepositERC20.js");
const writeDepositETH_js_1 = require("../actions/wallet/L1/writeDepositETH.js");
const writeDepositTransaction_js_1 = require("../actions/wallet/L1/writeDepositTransaction.js");
const writeFinalizeWithdrawalTransaction_js_1 = require("../actions/wallet/L1/writeFinalizeWithdrawalTransaction.js");
const writeProveWithdrawalTransaction_js_1 = require("../actions/wallet/L1/writeProveWithdrawalTransaction.js");
function walletL1OpStackActions(client) {
    return {
        writeDepositTransaction: (args) => (0, writeDepositTransaction_js_1.writeDepositTransaction)(client, args),
        writeDepositETH: (args) => (0, writeDepositETH_js_1.writeDepositETH)(client, args),
        writeDepositERC20: (args) => (0, writeDepositERC20_js_1.writeDepositERC20)(client, args),
        writeProveWithdrawalTransaction: (args) => (0, writeProveWithdrawalTransaction_js_1.writeProveWithdrawalTransaction)(client, args),
        writeFinalizeWithdrawalTransaction: (args) => (0, writeFinalizeWithdrawalTransaction_js_1.writeFinalizeWithdrawalTranasction)(client, args),
    };
}
exports.walletL1OpStackActions = walletL1OpStackActions;
