// op-viem/dist/esm/types/opStackContracts.js
export var OpStackL1Contract;
(function (OpStackL1Contract) {
    OpStackL1Contract["L1CrossDomainMessenger"] = "l1CrossDomainMessenger";
    OpStackL1Contract["L1Erc721Bridge"] = "l1Erc721Bridge";
    OpStackL1Contract["L1StandardBridge"] = "l1StandardBridge";
    OpStackL1Contract["L2OutputOracle"] = "l2OutputOracle";
    OpStackL1Contract["OptimismPortal"] = "optimismPortal";
})(OpStackL1Contract || (OpStackL1Contract = {}));
export var OpStackL2Contract;
(function (OpStackL2Contract) {
    OpStackL2Contract["L2CrossDomainMessenger"] = "l2CrossDomainMessenger";
    OpStackL2Contract["L2StandardBridge"] = "l2StandardBridge";
    OpStackL2Contract["GasPriceOracle"] = "gasPriceOracle";
    OpStackL2Contract["L1Block"] = "l1Block";
    OpStackL2Contract["L2ToL1MessagePasser"] = "l2ToL1MessagePasser";
    OpStackL2Contract["L2Erc721Bridge"] = "l2Erc721Bridge";
    OpStackL2Contract["OptimismMintableErc721Factory"] = "optimismMintableErc721Factory";
})(OpStackL2Contract || (OpStackL2Contract = {}));
export const opStackL2ChainContracts = {
    l2CrossDomainMessenger: { address: '0x4200000000000000000000000000000000000007' },
    l2StandardBridge: { address: '0x4200000000000000000000000000000000000010' },
    gasPriceOracle: { address: '0x420000000000000000000000000000000000000F' },
    l1Block: { address: '0x4200000000000000000000000000000000000015' },
    l2ToL1MessagePasser: { address: '0x4200000000000000000000000000000000000016' },
    l2Erc721Bridge: { address: '0x4200000000000000000000000000000000000014' },
    optimismMintableErc721Factory: { address: '0x4200000000000000000000000000000000000017' },
};
