// op-viem/dist/cjs/actions/wallet/L1/writeFinalizeWithdrawalTransaction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.writeFinalizeWithdrawalTranasction = exports.FUNCTION = exports.CONTRACT = exports.ABI = void 0;
const contracts_ts_1 = require("@eth-optimism/contracts-ts");
const opStackContracts_js_1 = require("../../../types/opStackContracts.js");
const writeOpStackL1_js_1 = require("./writeOpStackL1.js");
exports.ABI = contracts_ts_1.optimismPortalABI;
exports.CONTRACT = opStackContracts_js_1.OpStackL1Contract.OptimismPortal;
exports.FUNCTION = 'finalizeWithdrawalTransaction';
async function writeFinalizeWithdrawalTranasction(client, { withdrawal, optimismPortalAddress, ...rest }) {
    return (0, writeOpStackL1_js_1.writeOpStackL1)(client, {
        address: optimismPortalAddress,
        abi: exports.ABI,
        contract: exports.CONTRACT,
        functionName: exports.FUNCTION,
        args: [withdrawal],
        ...rest,
    });
}
exports.writeFinalizeWithdrawalTranasction = writeFinalizeWithdrawalTranasction;
