// op-viem/dist/esm/actions/wallet/L1/writeProveWithdrawalTransaction.js
import { optimismPortalABI } from '@eth-optimism/contracts-ts';
import { OpStackL1Contract } from '../../../types/opStackContracts.js';
import { writeOpStackL1 } from './writeOpStackL1.js';
export const ABI = optimismPortalABI;
export const CONTRACT = OpStackL1Contract.OptimismPortal;
export const FUNCTION = 'proveWithdrawalTransaction';
/**
 * Calls proveWithdrawalTransaction on the OptimismPortal contract.
 * Is the first L1 step of a withdrawal.
 *
 * @param parameters - {@link WriteProveWithdrawalTransactionParameters}
 * @returns A [Transaction Hash](https://viem.sh/docs/glossary/terms.html#hash). {@link WriteContractReturnType}
 */
export async function writeProveWithdrawalTransaction(client, { args: { withdrawalTransaction, outputRootProof, withdrawalProof, L2OutputIndex }, optimismPortalAddress, ...rest }) {
    return writeOpStackL1(client, {
        address: optimismPortalAddress,
        abi: ABI,
        contract: CONTRACT,
        functionName: FUNCTION,
        args: [withdrawalTransaction, L2OutputIndex, outputRootProof, withdrawalProof],
        ...rest,
    });
}
