// op-viem/dist/esm/decorators/publicL1OpStackActions.js
import { getL2HashesForDepositTx, } from '../actions/public/L1/getL2HashesForDepositTx.js';
import { getOutputForL2Block, } from '../actions/public/L1/getOutputForL2Block.js';
import { getSecondsToFinalizable, } from '../actions/public/L1/getSecondsToFinalizable.js';
import { getSecondsToNextL2Output, } from '../actions/public/L1/getSecondsToNextL2Output.js';
import { readFinalizedWithdrawals, } from '../actions/public/L1/readFinalizedWithdrawals.js';
import { readOpStackL1 } from '../actions/public/L1/readOpStackL1.js';
import { readProvenWithdrawals, } from '../actions/public/L1/readProvenWithdrawals.js';
import { simulateDepositERC20, } from '../actions/public/L1/simulateDepositERC20.js';
import { simulateDepositETH, } from '../actions/public/L1/simulateDepositETH.js';
import { simulateProveWithdrawalTransaction, } from '../actions/public/L1/simulateProveWithdrawalTransaction.js';
export function publicL1OpStackActions(client) {
    return {
        getL2HashesForDepositTx: (args) => getL2HashesForDepositTx(client, args),
        simulateDepositETH: (args) => simulateDepositETH(client, args),
        simulateDepositERC20: (args) => simulateDepositERC20(client, args),
        getOutputForL2Block: (args) => getOutputForL2Block(client, args),
        simulateProveWithdrawTransaction: (args) => simulateProveWithdrawalTransaction(client, args),
        getSecondsToNextL2Output: (args) => getSecondsToNextL2Output(client, args),
        getSecondsToFinalizable: (args) => getSecondsToFinalizable(client, args),
        readProvenWithdrawals: (args) => readProvenWithdrawals(client, args),
        readFinalizedWithdrawals: (args) => readFinalizedWithdrawals(client, args),
        readOpStackL1: (args) => readOpStackL1(client, args),
    };
}
